apps/history/declarativehistorymodel.cpp

```diff
@@ -38,6 +42,7 @@ void DeclarativeHistoryModel::clear()
     m_links.clear();
     endResetModel();
     DBManager::instance()->clearHistory();
+    FaviconManager::instance()->clear(QStringLiteral("history"));
```

Comment:
Following testing I notice that `DeclarativeHistoryModel::clear()` [doesn't actually get called](https://github.com/sailfishos/sailfish-browser/blob/84fffac3ec4fe1e4ddbfbb1babee4b3b561ae32a/apps/core/settingmanager.cpp#L104) when the history is cleared.

For passwords the icons are [cleared directly](https://github.com/sailfishos/sailfish-browser/blob/84fffac3ec4fe1e4ddbfbb1babee4b3b561ae32a/apps/core/settingmanager.cpp#L125) in the `SettingsManager`, which might be one way to address this. But if something more complex is needed this could be lodged as a separate bug.

Reply:
Interesting, nice catch.. I have created JB#54848 / OMP#JOLLA-230 to track this.  I guess that given that there are different data types (passwords, history items, favicons) and both in-memory and in-database copies of the data, there might be some complexities / side effects which need to be considered..

Reply:
Thanks for looking at this and for creating the bug.